src/features/terminal/shells/pwsh/pwshStartup.ts

```diff
@@ -57,6 +57,27 @@ async function isPowerShellInstalled(shell: string): Promise<boolean> {
     }
 }
 
+/**
+ * Detects the major version of PowerShell by executing a version query command.
+ * This helps with debugging activation issues since PowerShell 5.x and 7+ have different behaviors.
+ * @param shell The PowerShell executable name ('powershell' for Windows PowerShell or 'pwsh' for PowerShell Core/7+)
+ * @returns Promise resolving to the major version number as a string, or undefined if detection fails
+ */
+async function getPowerShellVersion(shell: 'powershell' | 'pwsh'): Promise<string | undefined> {
+    try {
+        const command = `${shell} -Command "\$PSVersionTable.PSVersion.Major"`;
```

Comment:
This looks good, I think you can even make it `pwsh -c '$PSVersionTable.PSVersion.Major'`